truffle/src/main/java/org/jruby/truffle/nodes/core/ReturnEnumeratorIfNoBlockNode.java

```diff
@@ -41,4 +43,12 @@ public Object execute(VirtualFrame frame) {
         }
     }
 
+    private static <T> T[] unshift(T first, T[] array) {
```

Comment:
Yeah - feel free to move this helper into RubyArguments and make something like `extractUserArgumentsWithUnshift`.